rspace/src/main/scala/coop/rchain/rspace/merger/StateChangeMerger.scala

```diff
@@ -132,8 +132,7 @@ object StateChangeMerger {
                       for {
                         joinsPointer <- joinsPointerForChannel(channel)
                         joins        <- baseReader.getJoins(joinsPointer).map(_.map(_.raw))
-
-                      } yield (channel, joins)
+                      } yield (joinsPointer, joins)
```

Comment:
This was a bug in merging joins - instead of using pointer to joins (from channels map) straight channel has been used.

Reply:
Nice catch!